src/routes/write-exercise/components/WriteExercise.tsx

```diff
@@ -41,14 +42,23 @@ const WriteExercise = ({ documents, route, navigation }: WriteExercisePropType):
     }))
   )
 
+  const keyboard = useKeyboard()
   const current = documentsWithResults[currentIndex]
   const nthRetry = current.numberOfTries ?? 0
   const needsToBeRepeated = nthRetry < numberOfMaxRetries && current.result !== SIMPLE_RESULTS.correct
 
   const tryLater = useCallback(() => {
-    setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
-    Keyboard.dismiss()
-  }, [documentsWithResults, currentIndex])
+    // Wait until keyboard is hidden before go next
```

Comment:
ImageViewer component does not resize properly if we do not wait for the keyboard to be dismissed before switching to the next question

Reply:
Why do we only need to handle this for `tryLater` and not the other buttons like check and give up? Also comments should better tell WHY something is done than WHAT is done, the what should be clear most of the time by looking at the code (oc not always possible).
```suggestion
    // ImageViewer is not resized correctly if keyboard is not dismissed before going to next document
```